compiler/rustc_pattern_analysis/src/constructor.rs

```diff
@@ -681,6 +681,8 @@ pub enum Constructor<Cx: PatCx> {
     Variant(Cx::VariantIdx),
     /// References
     Ref,
+    /// Pinned references (enabled by the `pin_ergonomics` feature)
+    RefPin,
```

Comment:
Here `RefPin` means `&pin const $pat` or `&pin mut $pat` which turns `&pin mut T` or `&pin const T` from by-pinned-ref binding mode to by-value mode. As mentioned in [#139751 (issue comment)](https://github.com/rust-lang/rust/pull/139751#discussion_r2062606467).

I'm not sure if the name is OK.